2-Operating-Systems/1-arch.md

```diff
@@ -0,0 +1,275 @@
+    ```
+
+    Explanation:
+    ```
+    sgdisk \
+       -Zo             `# zero out any existing partitions` \
+       -n 1:2048:+512M `# create new 512M partition for boot` \
+       -t 1:EF00       `# set type to EFI` \
+       -c 1:boot       `# set comment/label to "boot"` \
+       -N 2            `# create second new partition filling rest of disk` \
+       -t 2:8300       `# set type Linux` \
+       -c 2:root       `# set comment/label to "root"` \
+       /dev/mmcblk0     # writing to your target drive
+     ```
+
+    Note: As an alternative, you can graphically partition using `cgdisk`
+
+  3. Format the EFI boot partition as FAT32
+
+     ```
+     mkfs.vfat -F32 /dev/mmcblk0p1
+     ```
+
+  4. Encrypt and format the root partition
+
+     ```
+     # encrypt root partition with passphrase
+     cryptsetup -y -v luksFormat --type luks2 /dev/mmcblk0p2
+
+     # decrypt drive and expose as /dev/mapper/cryptroot
+     cryptsetup open /dev/mmcblk0p2 cryptroot
+
+     # make journaled ext4 partition in decrypted root device
+     mkfs.ext4 -j /dev/mapper/cryptroot
+     ```
+
+8. Mount the file systems
+
+  ```
+  mount /dev/mapper/cryptroot /mnt
+  mkdir /mnt/boot
+  mount /dev/mmcblk0p1 /mnt/boot
+  ```
+
+9. Select the Mirrors
+
+  ```
+  pacman -Sy pacman-contrib
+  curl https://www.archlinux.org/mirrorlist/all/https/ \
+  | sed 's/^#Server/Server/g' \
+  | rankmirrors -n 0 - \
+  > /etc/pacman.d/mirrorlist
+  ```
+  Note: This avoids manual mirror sorting. Native script does not yet exist.
+
+10. Install the base packages
+
+  ```
+  pacstrap /mnt base
+  ```
+
+11. Configure the System
+  1. Fstab: Generate 'File System TABle' of contents
+
+  ```
+  genfstab -U /mnt >> /mnt/etc/fstab
+  ```
+
+  2. Chroot: login to new arch installation by changing root
+
+  ```
+  arch-chroot /mnt
+  ```
+
+  3. Set time zone
+
+  ```
+  timedatectl set-timezone Region/City
+  hwclock --systohc
+  ```
+
+  4. Localization: setting your preferred language
+
+  ```
+  # Uncomment preferred language in /etc/locale.gen
+  sed -i 's/^#en_US/en_US/g' /etc/locale.gen
+
+  # Create /etc/locale.conf and set default language
+  echo "LANG=en_US.UTF-8" > /etc/locale.conf
+
+  # Non-US keyboard users can set their default layout
+  echo "KEYMAP=de-latin1" > /etc/vconsole.conf
+  ```
+
+  5. Name your computer!
+
+  ```
+  # Create the hostname file
+  hostnamectl set-hostname computername
+
+  # Specify new hostname for the network
+  echo "127.0.0.1 computername.localdomain computername localhost" > /etc/hosts
+  echo "::1 computername.localdomain computername localhost" >> /etc/hosts
+  ```
+
+  6. Initramfs: install system initialization bundle
+
+  ```
+  # Enable encryption support in initramfs
+  sed -i 's/ filesystems/ encrypt filesystems/g' /etc/mkinitcpio.conf
+
+  # Generate new initramfs
+  mkinitcpio -p linux
+  ```
+
+  7. Create a user with super user rights
+
+  ```
+  pacman -S sudo
```

Comment:
Should add `sudo` to your pacstrap command above so that it all gets done in one step.